libclc/ptx-nvidiacl/libspirv/atomic/atomic_helpers.h

```diff
@@ -72,6 +72,15 @@ _CLC_OVERLOAD _CLC_DECL void __spirv_MemoryBarrier(unsigned int, unsigned int);
   }                                                                            \
   }
 
+#define __CLC_NVVM_FENCE_SC_SM70()                                             \
+  if (scope == CrossDevice) {                                                  \
+    __asm__ __volatile__("fence.sc.sys;");                                     \
+  } else if (scope == Device) {                                                \
+    __asm__ __volatile__("fence.sc.gpu;");                                     \
+  } else {                                                                     \
+    __asm__ __volatile__("fence.sc.cta;");                                     \
```

Comment:
I think that the else state here assumes also `scope::Invocation` (mapping from `sycl::memory_scope::work_item`) as well as the intended for `Workgroup` and `Subgroup` for the `cta` fence. Maybe can do with the explicit scope check for just equal Workgroup/Subgroup, or a switch-statement like in the above macros that handle scope. Not a big deal, just checking if intended.

Reply:
Yeah this is intended. `sycl::memory_scope::work_item` also maps to cuda's `thread_group`, which uses `fence.sc.cta`.

Reply:
I see, thanks! I should more docs.. Nice work!